tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,496 @@
+# Spatially smooth with a 2D Gaussian kernel
+size = width // 2 - 1
+gaussian = np.exp(-(np.arange(-size, size + 1) ** 2 / float(gaussian_sd ** 2)))
+for si in range(X.shape[0]):
+    for ri in range(X.shape[1]):
+        X[si, ri, :] = np.convolve(X[si, ri, :], gaussian, 'same')
+    for ci in range(X.shape[2]):
+        X[si, :, ci] = np.convolve(X[si, :, ci], gaussian, 'same')
+
+###############################################################################
+# In this case, a null hypothesis we could test is for each voxel is:
+#
+#     There is no difference between the mean value and zero
+#     (:math:`H_0: \mu = 0`).
+#
+# The alternative hypothesis, then, is that the voxel has a non-zero mean.
+# This is a *two-tailed test* because this means the mean could be less than
+# or greater-than zero (whereas a *one-tailed test* would test only one of
+# these possibilities, i.e. :math:`H_0: \mu \geq 0` or
+# :math:`H_0: \mu \leq 0`).
+#
+# Parametric tests
+# ^^^^^^^^^^^^^^^^
+# Let's start with a **1-sample t-test**, which is a standard test
+# for differences in paired sample means. This test is **parametric**,
+# as it assumes that the underlying sample distribution is Gaussian, and is
+# only valid in this case. (This happens to be satisfied by our toy dataset,
+# but is not always satisfied for neuroimaging data.)
+#
+# In the context of our toy dataset, which has many voxels, applying the
+# 1-sample t test is called a *mass-univariate* approach as it treats
+# each voxel independently.
+
+# X needs to be a multi-dimensional array of shape (n_samples, n_time, n_space)
+# (where samples here are subjects) so we permute dimensions:
+t_uncorrected, p_uncorrected = stats.ttest_1samp(X, 0, axis=0)
+
+###############################################################################
+# "hat" variance adjustment
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# The "hat" technique regularizes the variance values used in the t-test
+# calculation [1]_ to compensate for implausibly small variances.
+t_uncorrected_hat = ttest_1samp_no_p(X, sigma=sigma)
+p_uncorrected_hat = stats.distributions.t.sf(
+    np.abs(t_uncorrected_hat), len(X) - 1) * 2
+
+###############################################################################
+# Non-parametric tests
+# ^^^^^^^^^^^^^^^^^^^^
+# Instead of assuming an underlying Gaussian distribution, we could instead
+# use a **non-parametric permutation** method. Under the null hypothesis,
+# we have the princple of **exchangability**, which means that, if the null
+# is true, we should be able to exchange conditions and not change the
+# distribution of the test statistic.
+#
+# In the case of a paired t test against 0 (or between two conditions where
+# you have already subtracted them), exchangeability means that we can flip
+# the signs of our data. Therefor we can bootstrap the **null distribution**
+# values by taking random subsets of samples (subjects), flipping the sign
```

Comment:
Related to my comment before, I think we're not taking random subsamples in permutation testing, but we're using all available samples and flip signs.